File: src/main/java/io/github/saneea/feature/text/AddNewLine.java
package io.github.saneea.feature.text;

import java.io.IOException;
import java.io.Reader;
import java.io.Writer;

import io.github.saneea.Feature;
import io.github.saneea.FeatureContext;

public class AddNewLine implements//
		Feature, //
		Feature.In.Text.Reader, //
		Feature.Out.Text.Writer {

	private Reader in;
	private Writer out;

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("add platform-dependent 'newline' sequence (e.g. '\\n', '\\r\\n') at the end");
	}

	@Override
	public void run(FeatureContext context) throws IOException {
		in.transferTo(out);
		out.append(System.lineSeparator());
	}

	@Override
	public void setIn(Reader in) {
		this.in = in;
	}

	@Override
	public void setOut(Writer out) {
		this.out = out;
	}

}

File: src/main/kotlin/io/github/saneea/dvh/feature/text/AddNewLine.kt
package io.github.saneea.dvh.feature.text

import io.github.saneea.dvh.Feature
import io.github.saneea.dvh.Feature.Meta
import java.io.Reader
import java.io.Writer

class AddNewLine :
    Feature,
    Feature.In.Text.Reader,
    Feature.Out.Text.Writer {

    override lateinit var inTextReader: Reader
    override lateinit var outTextWriter: Writer

    override val meta = Meta("add platform-dependent 'newline' sequence (e.g. '\\n', '\\r\\n') at the end")

    override fun run() {
        inTextReader.transferTo(outTextWriter)
        outTextWriter.append(System.lineSeparator())
    }
}